QModManager/API/SMLHelper/Handlers/KnownTechHandler.cs

```diff
@@ -0,0 +1,171 @@
+﻿namespace QModManager.API.SMLHelper.Handlers
+{
+    using Interfaces;
+    using Patchers;
+    using System.Collections.Generic;
+    using System.Linq;
+    using UnityEngine;
+
+    /// <summary>
+    /// A handler class for configuring custom unlocking conditions for item blueprints.
+    /// </summary>
+    public class KnownTechHandler : IKnownTechHandler
+    {
+        private static readonly KnownTechHandler singleton = new KnownTechHandler();
+
+        public static IKnownTechHandler Main => singleton;
```

Comment:
Some handlers only have `Main`, others have both `singleton` and `Main`.

Reply:
This was necessary as in some of the handlers it was necessary to access the actual class backing the public interface.  
For those handlers where everything could be done with just the public interface, `Main` alone was kept.  
For handlers where we needed to keep a private instance of the actual class, `singleton` was used.

Reply:
Alright. Some are `singleton` and some are `Singleton` though (I have not been diagnosed with OCD (yet))